tiledb/sm/c_api/tiledb.cc

```diff
@@ -4274,6 +4275,43 @@ int32_t tiledb_serialize_array_max_buffer_sizes(
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_deserialize_fragments_timestamps(
+    tiledb_ctx_t*,
+    tiledb_serialization_type_t serialize_type,
+    const tiledb_buffer_t* buffer,
+    uint64_t* timestamp_start,
+    uint64_t* timestamp_end) noexcept {
+  api::ensure_buffer_is_valid(buffer);
+
+  // Deserialize
+  tie(*timestamp_start, *timestamp_end) =
+      tiledb::sm::serialization::fragments_timestamps_deserialize(
+          (tiledb::sm::SerializationType)serialize_type, buffer->buffer());
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_deserialize_fragments_list(
+    tiledb_ctx_t*,
+    const char* array_uri,
+    tiledb_serialization_type_t serialize_type,
+    const tiledb_buffer_t* buffer,
+    tiledb_fragments_list_t** fragments) noexcept {
+  api::ensure_buffer_is_valid(buffer);
```

Comment:
What happens if `array_uri` is `nullptr`?

Reply:
If `array_uri` is `nullptr`, the default constructor of `URI` will create an empty uri (`""`). Thus, the internal validation check [here](https://github.com/TileDB-Inc/TileDB/pull/3923/files#diff-f367f53e463b652870ff3932e7a5203ccaafc3bda6f677a19839112ca180db3cR565-R568) will fail as anticipated.

Reply:
Do we have a test for that?